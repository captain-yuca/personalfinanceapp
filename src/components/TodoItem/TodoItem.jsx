import React from 'react'
import PropTypes from 'prop-types'

import { Wrapper, Text, Remove } from './TodoItem.styles'

const propTypes = {
  todo: PropTypes.shape({
    text: PropTypes.string,
  }),
  removeTodo: PropTypes.func.isRequired,
}

function TodoItem({ todo, removeTodo }) {
  return (
    <Wrapper>
      <Text>{todo.text}</Text>
      <Remove onClick={() => removeTodo(todo)}>X</Remove>
    </Wrapper>
  )
}

TodoItem.propTypes = propTypes

export default TodoItem
