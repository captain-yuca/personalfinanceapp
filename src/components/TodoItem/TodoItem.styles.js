import styled from 'styled-components'

export const Text = styled.span`
  color: ${({ theme }) => theme.colors.text};
  font-size: 14px;
  transition: all 0.3s ease;
`

export const Remove = styled.span`
  color: ${({ theme }) => theme.colors.text};
  font-weight: bold;
  transition: all 0.3s ease;
`

export const Wrapper = styled.div`
  border-bottom: 2px solid ${({ theme }) => theme.colors.primary};
  padding: 8px 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  &:hover {
    ${Text}, ${Remove} {
      color: ${({ theme }) => theme.colors.primary};
    }
  }
`
