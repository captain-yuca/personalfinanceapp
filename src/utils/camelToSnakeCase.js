export const camelToSnakeCase = str => str.replace(/([A-Z])/g, group => `_${group.toLowerCase()}`)
