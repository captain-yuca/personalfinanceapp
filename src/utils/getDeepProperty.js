export const getDeepProperty = (path, obj) =>
  path.split('.').reduce((prev, curr) => (prev ? prev[curr] : null), obj || null)
