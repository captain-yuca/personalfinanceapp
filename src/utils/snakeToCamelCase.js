export const snakeToCamelCase = str =>
  str.replace(/([-_][a-z])/g, group => group.toUpperCase().replace('-', '').replace('_', ''))
