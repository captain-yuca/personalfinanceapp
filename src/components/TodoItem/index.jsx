import React from 'react'
import TodoItem from './TodoItem'

function Wrapper(props) {
  return <TodoItem {...props} />
}

export default Wrapper
