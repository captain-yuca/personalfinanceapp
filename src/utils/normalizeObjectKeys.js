import { camelToSnakeCase } from './camelToSnakeCase'
import { snakeToCamelCase } from './snakeToCamelCase'

export const normalizeObjectKeys = (obj, toSnake = false) =>
  Object.entries(obj).reduce((accum, [rawKey, value]) => {
    const key = toSnake ? camelToSnakeCase(rawKey) : snakeToCamelCase(rawKey)
    return { ...accum, [key]: value }
  }, {})
