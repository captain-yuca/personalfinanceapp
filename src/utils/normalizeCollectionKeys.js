import { normalizeObjectKeys } from './normalizeObjectKeys'

export const normalizeCollectionKeys = (collection, toSnake = false) =>
  collection.map(item => normalizeObjectKeys(item, toSnake))
