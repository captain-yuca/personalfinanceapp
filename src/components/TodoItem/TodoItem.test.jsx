import React from 'react'
import { render, fireEvent } from '../../test.utils'

import TodoItem from './TodoItem'
import TodoWrapper from '.'

describe('TodoItem component', () => {
  const props = {
    todo: {
      text: 'TODO',
    },
    removeTodo: jest.fn(),
  }

  it('should render correctly', () => {
    const { container, getByText } = render(<TodoItem {...props} />)

    expect(getByText(props.todo.text)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should correctly called removeTodo after click on x btn', () => {
    const { getByText } = render(<TodoItem {...props} />)

    const btn = getByText('X')
    fireEvent.click(btn)

    expect(props.removeTodo).toBeCalledWith(props.todo)
  })
})

describe('TodoWrapper component', () => {
  const props = {
    todo: {
      text: 'TODO',
    },
    removeTodo: jest.fn(),
  }

  it('should render correctly', () => {
    const { container } = render(<TodoWrapper {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('should handle click on add btn', () => {
    const { getByText } = render(<TodoWrapper {...props} />)

    const btn = getByText('X')
    fireEvent.click(btn)
  })
})
