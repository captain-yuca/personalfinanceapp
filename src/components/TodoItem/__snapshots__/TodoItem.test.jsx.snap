// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem component should render correctly 1`] = `
.c2 {
  color: #ffffff;
  font-size: 14px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c4 {
  color: #ffffff;
  font-weight: bold;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0 {
  border-bottom: 2px solid #ff0000;
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.c0:hover .c1,
.c0:hover .c3 {
  color: #ff0000;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
    >
      TODO
    </span>
    <span
      class="c3 c4"
    >
      X
    </span>
  </div>
</div>
`;

exports[`TodoWrapper component should render correctly 1`] = `
.c2 {
  color: #ffffff;
  font-size: 14px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c4 {
  color: #ffffff;
  font-weight: bold;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0 {
  border-bottom: 2px solid #ff0000;
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.c0:hover .c1,
.c0:hover .c3 {
  color: #ff0000;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
    >
      TODO
    </span>
    <span
      class="c3 c4"
    >
      X
    </span>
  </div>
</div>
`;
